app/src/shared/components/PasswordStrength/index.jsx

```diff
@@ -0,0 +1,86 @@
+// @flow
+
+import { PureComponent, type Node } from 'react'
+
+import CancelledPromiseError from '$shared/errors/CancelledPromiseError'
+import makeCancelable, { type Cancelable } from '$utils/makeCancelable'
+import zxcvbn from '$utils/zxcvbn'
+
+type Props = {
+    children?: (number) => Node,
+    enabled?: boolean,
+    value?: string,
+}
+
+type State = {
+    measurer: ?(string) => {
+        score: number,
+    },
+    strength: number,
+}
+
+class PasswordStrength extends PureComponent<Props, State> {
+    state = {
+        measurer: null,
+        strength: -1,
+    }
+
+    componentDidMount() {
+        this.loadMeasurer()
+    }
+
+    componentDidUpdate() {
+        this.loadMeasurer()
+        this.measure()
+    }
+
+    componentWillUnmount() {
+        const { getZxcvbn } = this
+        if (getZxcvbn) {
+            getZxcvbn.cancel()
+        }
+    }
+
+    getZxcvbn: ?Cancelable = null
+
+    loadMeasurer() {
+        const { enabled, value } = this.props
+
+        if (!this.getZxcvbn && enabled && value) {
+            this.getZxcvbn = makeCancelable(zxcvbn())
+            this.getZxcvbn.promise.then((measurer) => {
+                this.setState({
+                    measurer,
+                }, this.measure)
+            }, (error) => {
+                if (!(error instanceof CancelledPromiseError)) {
+                    throw error
+                }
+            })
```

Comment:
Making a cancellable promise is cool… arguably more "correct" but it seems like a lot of mucking around compared to just setting a flag in `componentWillUnmount`:

```js
componentWillUnmount() {
    this._unmounted = true
}

loadMeasurer() {
    const { enabled, value } = this.props
    if (!enabled || !value) { return }
    const measurer = await zxcvbn()
    if (this._unmounted) { return }
    this.setState({ measurer }, this.measure)
} 
```

Cancelling is the right approach if there's something that can actually be cancelled, e.g. an expensive task, not sure it's the right option here.